tools/psa/build_tfm.py

```diff
@@ -160,22 +182,334 @@ def clean_up_cloned_repos():
+                debug = True
+
+            retcode = run_cmake_build(True, cmake_build_dir, tgt[4],
+                                        tgt[1], tgt[2], debug)
+            if retcode:
+                msg = "Cmake configure failed for target %s with toolchain %s" % (
+                                                        tgt[0], tgt[2])
+                logger.critical(msg)
+                sys.exit(1)
+            retcode = run_cmake_build(False, cmake_build_dir)
+            if retcode:
+                msg = "Cmake build failed for target %s using toolchain %s" % (
+                                                    tgt[0],  tgt[2])
+                logger.critical(msg)
+                sys.exit(1)
+
+            output_dir = None
+            if(tgt[3].endswith('/')):
+                output_dir = tgt[3]
+            else:
+                output_dir = tgt[3] + '/'
+
+            tfm_secure_bin = join(cmake_build_dir, 'install', 'outputs'
+                                    ,tgt[1], 'tfm_s.bin')
+            logger.info("Copying %s to %s" % (relpath(tfm_secure_bin,
+                                    ROOT), relpath(output_dir, ROOT)))
+            shutil.copy2(tfm_secure_bin, output_dir)
+            tfm_secure_bin = join(cmake_build_dir, 'install', 'outputs'
+                                    , tgt[1], 'tfm_s.axf')
+            logger.info("Copying %s to %s" % (relpath(tfm_secure_bin,
+                                    ROOT), relpath(output_dir, ROOT)))
+            shutil.copy2(tfm_secure_bin, output_dir)
+
+            if args.commit:
+                commit_changes(tgt[3], tgt[0], tgt[2])
+
+def clean_build_directory(build_dir):
+    cmake_build_dir = join(build_dir, 'trusted-firmware-m', 'cmake_build')
+    if isdir(cmake_build_dir):
+        logger.info("Removing Cmake build directory %s" %
+                                                    relpath(cmake_build_dir))
+        shutil.rmtree(cmake_build_dir)
+
+def get_parser():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("-m", "--mcu",
+                        help="Build for the given MCU",
+                        default=None,
+                        choices=get_tfm_secure_targets())
+    hmsg = "Build for the given toolchain (default is tfm_default_toolchain)"
+    parser.add_argument("-t", "--toolchain",
+                        help=hmsg,
+                        default=None,
+                        choices=["ARMCLANG", "GNUARM"])
+
+    parser.add_argument("-d", "--debug",
+                        help="Set build profile to debug",
+                        action="store_true",
+                        default=False)
+
+    parser.add_argument('-l', '--list',
+                        action="store_true",
+                        default=False,
+                        help="Print supported TF-M secure targets")
+
+    parser.add_argument("--commit",
+                        help="Create a git commit for each platform",
```

Comment:
Are we planning to support multiple commits for all supported platforms if `python3 tools/psa/build_tfm.py` used? It would be good to restrict this option only when a specific target is used?

Reply:
@jainvikas8 Good question.

@jainvikas8 @Patater 
While building tf-m for all targets, should we have commits for tf-m binary for every target (This will lead to `N` number of commits for every release where `N` is number of supported targets) or just one commit with tf-m binaries for all the targets included?

Reply:
One commit is fine. The aim of the commit would be "Update TF-M to version blah blah" and would include updating all necessary headers and binaries.